ampd/src/mvx/proxy.rs

```diff
@@ -6,11 +6,57 @@ use hex::ToHex;
 use mockall::automock;
 use multiversx_sdk::data::transaction::TransactionOnNetwork;
 use multiversx_sdk::gateway::GatewayProxy;
+use tracing::warn;
 
+use crate::monitoring;
+use crate::monitoring::metrics::Msg;
 use crate::types::Hash;
 
 const STATUS_SUCCESS: &str = "success";
 
+#[derive(Debug)]
+pub struct Client {
+    client: GatewayProxy,
+    monitoring_client: monitoring::Client,
+    handler_label: String,
+}
+
+impl Client {
+    pub fn new(
+        client: GatewayProxy,
+        monitoring_client: monitoring::Client,
+        handler_label: String,
+    ) -> Self {
+        Self {
+            client,
+            monitoring_client,
+            handler_label,
+        }
+    }
+
+    /// client.get_transaction_info_with_results returns anyhow::Error for TransactionNotFound.
+    /// Therefore, TransactionNotFound errors will also be tracked and logged.
+    async fn get_transaction_info_with_results(
+        &self,
+        tx_hash: &str,
+    ) -> anyhow::Result<TransactionOnNetwork> {
+        self.client
+            .get_transaction_info_with_results(tx_hash)
+            .await
+            .inspect_err(|err| {
+                warn!(
+                    handler_label = &self.handler_label,
+                    "received RPC error: {:?}", err
+                );
+                self.monitoring_client
+                    .metrics()
+                    .record_metric(Msg::RpcError {
+                        handler_label: self.handler_label.clone(),
+                    });
```

Comment:
just do this inside `transaction_info_with_results` and you don't even have to import anyhow